test/accountSpec.js

```diff
@@ -418,6 +417,42 @@ describe('Accounts', function () {
         .expect(expected)
         .end()
     })
+
+    it('should lowercase account name -- uppercased in url', function * () {
+      const lowerCased = this.exampleAccounts.bob
+      const account = _.assign({}, lowerCased, {id: lowerCased.id.toUpperCase()})
+      const withoutPassword = _.omit(lowerCased, 'password')
+
+      yield this.request()
+        .put(account.id)
+        .auth('admin', 'admin')
+        .send(account)
+        .expect(201)
+        .expect(withoutPassword)
+        .expect(validator.validateAccount)
+        .end()
+
+      // Check balances
+      expect((yield Account.findByName('bob')).getDataExternal()).to.deep.equal(withoutPassword)
+    })
+
+    it('should lowercase account name -- uppercased in body', function * () {
```

Comment:
Please add a test case like this that omits the `id` field from the body; I think it will fail